crates/net/src/http/request.rs

```diff
@@ -279,12 +279,12 @@ impl Request {
     /// Reads the request to completion, parsing it as JSON.
     #[cfg(feature = "json")]
     #[cfg_attr(docsrs, doc(cfg(feature = "json")))]
-    pub async fn json<T: DeserializeOwned>(&self) -> Result<T, Error> {
+    pub async fn json<T: DeserializeOwned>(self) -> Result<T, Error> {
         serde_json::from_str::<T>(&self.text().await?).map_err(Error::from)
     }
 
     /// Reads the reqeust as a String.
-    pub async fn text(&self) -> Result<String, Error> {
+    pub async fn text(self) -> Result<String, Error> {
```

Comment:
This change, while good in theory, also makes the API unnecessarily restrictive.

![image](https://github.com/rustwasm/gloo/assets/47357913/25c951f0-f55c-4ade-800f-d515ef15b312)

Notice how the object can still be used after `text()` is called